File: vscode/src/model/reviewModel.ts
export interface ReviewComment {
  id: string;
  line?: number;
  lineEnd?: number;
  category?: "Q" | "R";
  body: string;
}
export interface ReviewFile {
  file: string;
  comments: ReviewComment[];
}
export type ReviewStatus = "open" | "submitted";
export interface ReviewState {
  status: ReviewStatus;
  files: ReviewFile[];
}

File: intellij/src/main/kotlin/io/github/ompreview/model/ReviewModel.kt
package io.github.ompreview.model

data class ReviewComment(
    val id: String,
    val line: Int? = null,
    val lineEnd: Int? = null,
    val category: CommentCategory = CommentCategory.REQUEST,
    val body: String,
)

data class ReviewFile(
    val file: String,
    val comments: List<ReviewComment> = emptyList(),
)

enum class ReviewStatus {
    OPEN,
    SUBMITTED,
}

enum class CommentCategory {
    QUESTION,
    REQUEST,
}

data class ReviewState(
    val status: ReviewStatus = ReviewStatus.OPEN,
    val files: List<ReviewFile> = emptyList(),
)
